packages/serverpod/lib/src/server/message_central.dart

```diff
@@ -50,15 +44,19 @@ class MessageCentral {
       var channel = _channels[channelName];
       if (channel == null) return;
 
-      assert(
-          ignoreTypeIncompatibility ||
-              message.runtimeType == channel.messageType,
-          'The channel $channelName handles ${channel.messageType} messages.\n'
-          'So a message of type ${message.runtimeType} should not be added to this channel.\n'
-          'In case you really know what you are doing, you can bypass this message by setting `ignoreTypeIncompatibility` to true.');
-
-      for (var sessionCallback in channel.callbacks) {
-        sessionCallback(message);
+      for (var sessionCallback in channel) {
+        try {
+          sessionCallback(message);
+        } on TypeError catch (_) {
+          // Ignore since the developer has added callbacks taking different types to this Channel.
+        } catch (e, stackTrace) {
+          sessionCallback.session.log(
+            'Failed to execute callback in channel $channelName',
+            exception: e,
+            level: LogLevel.error,
+            stackTrace: stackTrace,
+          );
+        }
```

Comment:
I'd have added this part. Should I readd it?

Reply:
Please make sure that you are pulling my changes before you make any new edits.